### ISSUE TITLE
Azure ratecard setting azureClientSecret does not update on changes to secret

### ISSUE THREAD

**Issue by Sean-Holcomb** [opencost/opencost#1331]
I have found that changing the azureClientSecret in the secret does not change the value. This manifests in rate card continuing to fail when it is changed from an incorrect value to a correct value and not failing when changed from a correct value to an incorrect value. This is in contrast with other ratecard configuration values which appear to change when the secret is edited. 
Related to ZD#https://kubecost.zendesk.com/agent/tickets/2120
and
https://github.com/kubecost/kubecost-cost-model/issues/853

**Comment by AjayTripathy** [comment]
Good find, thanks, but this probably belongs in the helm chart repo. https://github.com/kubecost/cost-analyzer-helm-chart/issues

**Comment by sylus** [comment]
Oh this might be the bug that is tripping me up. even though I entered the secret many many times and directly compared to my manifest json file I keep getting:

```sh
{"code":200,"status":"success","data":{"Rate Card API":{"name":"Rate Card API","available":false,"error":"azure.BearerAuthorizer#WithAuthorization: Failed to refresh the Token for request to https://management.azure.com/subscription
```

Is there any quick mitigations for this bug? @AjayTripathy @Sean-Holcomb 

I am 100% my service-key.json is correct as I have checked quite a few times. I directly compared the mounted secrets in cost-analyzer to my service-key.json file.

**Comment by Sean-Holcomb** [comment]
In the short term you can run the endpoint '/model/updateConfigByKey' with the body
```
 "azureSubscriptionID": "###",
 "azureClientID": "###",
 "azureClientSecret": "###",
"azureTenantID": "###"
```
A fix to the issue the config not properly updating from the secret should be available in 1.96
